sklearn/model_selection/_classification_threshold.py

```diff
@@ -931,6 +932,17 @@ def _fit(self, X, y, **params):
         best_idx = objective_scores.argmax()
         self.best_score_ = objective_scores[best_idx]
         self.best_threshold_ = decision_thresholds[best_idx]
+        if (
+            objective_scores.max() - objective_scores.min()
+            <= np.finfo(objective_scores.dtype).eps
+        ):
+            warn(
+                f"The objective metric {self.scoring!r} is constant at "
```

Comment:
This is a good idea. Might help some users. Also I like that it is a warning not an error.

Reply:
I updated the PR in a0d62c51d49c31178ed5e6c052120d10d3ae865d to keep the 0.5 threshold in that case. Using a extreme near-zero threshold would be introduce very weird / unexpectedly biased behavior. Better keep a more neutral behavior in such pathological situation.